### ISSUE TITLE
Users are not successfully being "mock" authenticated against Northstar

### ISSUE THREAD

**Issue by DFurnes** [DoSomethingArchive/legacy-website#6260]
#### Current Behavior

We're currently running "mock" authentication against Northstar's `auth/verify` endpoint with each user's credentials when they log in to Phoenix. This is a dry run for authenticating users against Northstar sometime in the future (rather than the local Phoenix DB).

Frustratingly, a lot of users (62%) are being authenticated against Phoenix but failing when we try to then use the same credentials for Northstar. We can see this because we save the user's Northstar ID to their profile on login if we can verify their credentials, but save `NONE` if something goes wrong.

The mock authentication code was added in #6196 and updated in #6198.
#### Desired Behavior

All users that can log in to Phoenix should be able to be authenticated against Northstar using the same credentials (i.e. we should never be failing and saving `NONE` to the Northstar ID field).
#### Steps to Replicate

To see users who failed the `auth/verify` step :

``` sql
SELECT * FROM field_data_field_northstar_id as n
INNER JOIN users u on u.uid = n.entity_id
WHERE n.field_northstar_id_value='NONE'
LIMIT 1000;
```
#### Why This Matters

One day, we should be authenticating all users against Northstar as our single "source of truth", rather than having two databases with overlapping user profiles. See #4526.

**Comment by angaither** [comment]
let's go the log all the things route. 

it's hard to find a discernible pattern in why this isn't working. 
make a `dosomething_user_northstar_log` table that can be turned on
we can log when a request is sent to both auth & update a user

store something like

| uid | op | user_lang | user country | json_encode($values) | northstar response | json_encode(ns data) |
| --- | --- | --- | --- | --- | --- | --- |
| 10 | auth | english | us | blah | 200 | blah |

**Comment by angaither** [comment]
bugs so far
- [ ] user failed with invalid credentials 
  - (uid 2881255) used mobile number to log in
  - uid 1506487 email
- [x] user failed with invalid email 
  - uid 2881255 (extra space after email) 

known & accepted
- a bunch of people are getting a 404 on registering, it seems the update hook is called before the register hook and the first request is failing b/c the user doesn't exist yet, but the registration hook passes fine.

**Comment by angaither** [comment]
we are also seeing quite a few user country's not saving, even though the fastly country code is set.

**Comment by DFurnes** [comment]
The issue with not removing the extra space was fixed in DoSomething/northstar#311. A bunch of other issues where duplicate user accounts could be created were fixed in DoSomething/northstar#309.

I tested logging in with my mobile number and it validated correctly, so my _guess_ is the remaining users are failing due to actual invalid credentials stored in Northstar. It also looks like any users who got a `401` on verifying their credentials are successfully verified with a `200` the next time they log in (since we update their password in Northstar after the first login).

Perhaps we were storing updated passwords on duplicate accounts that got merged, and so a subset of users ended up with the wrong credentials on their deduped account?

**Comment by angaither** [comment]
from @DFurnes 

> might be a good thing to re-run [northstar script](https://github.com/DoSomething/phoenix/blob/dev/scripts/export-users-to-northstar.php) to clear up our invalid credentials issues
> should also add `phoenix` as the source for the user if it's blank

**Comment by DFurnes** [comment]
We're currently re-running the migration, and things seem to be chugging along. :steam_locomotive: 

Out of 24,866 users migrated so far, 7,362 haven't been migrated because they have neither an email or password. And 125 users haven't been migrated because they have invalid email addresses (e.g. they have a space in their email, trailing period, completely random nonsense...)

See #6309 for more details on some of the "invalid" user data.

**Comment by DFurnes** [comment]
From checking out the failed request logs, I also noticed we had a bunch of "failures" that weren't _actually_ failures because we were trying to verify newly registered users _before_ their profiles were synced to Northstar (so we'd have a `401` followed up by a `200` sync in the logs).

This was confirmed by `dpm`ing when each hook and method occurred and fixed in #6401 above.

**Comment by DFurnes** [comment]
From updates in #6411, users with `1234445555@mobile` accounts can now be synced/migrated.
